code/modules/plumbing/plumbers/synthesizer.dm

```diff
@@ -0,0 +1,104 @@
+///A single machine that produces a single chem. Can be placed in unison with others through plumbing to create chemical factories
+/obj/machinery/plumbing/synthesizer
+	name = "chemical synthesizer"
+	desc = "Produces a single chemical at a given volume. Must be plumbed. Most effective when working in unison with other chemical synthesizers, heaters and filters."
+
+	icon_state = "synthesizer"
+	icon = 'icons/obj/plumbing/plumbers.dmi'
+
+	///Amount we produce for every process. Ideally keep under 5 since thats currently the standard duct capacity
+	var/amount = 1
+	///The maximum we can produce for every process
+	var/max_amount = 4
+	///I track them here because I have no idea how I'd make tgui loop like that
+	var/static/list/possible_amounts = list(1,2,3,4)
```

Comment:
shouldn't "5" be in this list?

Reply:
Ducts cant transfer more than 5 units at a time, so it would make mixing impossible. I guess there’s no harm in doing it anyway, in-case someone just wants retarted amounts of one chemical